lib/controller.js

```diff
@@ -170,6 +177,134 @@ Tessel.get = function(opts) {
     }
   });
 };
+/*
+Because of sync problem with master trunk and parallel development on Tessel.list 
+it was necessary to go my own way...
+
+The seekTessels method never uses heuristics because all possible login variants are required.
+There is a new feature in: In the case your network topology causes a Tessels IP is found twice, the 
+menu will only list it once.
+
+Due to my own stupid failing with mixing up a tessel is accessable via gateway and a Tessel is in the 
+same Netmask, I've added a notice for the case someone run into same issue.
+*/
+Tessel.seekTessels = function(opts) {
+  return new Promise(function(resolve, reject) {
+
+    if (opts.timeout && typeof opts.timeout === 'number') {
+      // Stop the search after that duration
+      setTimeout(stopSeeker, opts.timeout * 1000);
+    } else {
+      // default to 3 seconds searching
+      setTimeout(stopSeeker, 3000);
+    }
+    logs.info('Searching accessable Tessels ...');
+
+    var seeker = new discover.TesselSeeker().start();
+    var tessels = [];
+
+    // When we find Tessels
+    seeker.on('tessel', function(tessel) {
+      controller.closeTesselConnections(tessel)
+        .then(function() {
+          var known = false;
+          if (tessels.length >= 1) {
+            for (var i in tessels) {
```

Comment:
Don't use `for-in` to iterate arrays, use either regular `for` loop or `tessels.forEach`